### DISCUSSION TITLE
Manually or zero-code mongodb tracing with spring boot and micrometer

### DISCUSSION THREAD

**Discussion by jeusdi** [open-telemetry/opentelemetry-java-instrumentation#11173] [Q&A]
How can I configure Spring Boot 3 with Micrometer tracing to send traces to X-Ray instead of Jaeger, and also enable instrumentation to generate traces for communication with MongoDB, which I currently don't see visualized?

Here my current related dependencies I'm using:

```xml
<dependency>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-starter-actuator</artifactId>
</dependency>
<dependency>
		<groupId>io.micrometer</groupId>
		<artifactId>micrometer-tracing</artifactId>
</dependency>
<!-- Adds the Tracer Implementation -->
<dependency>
		<groupId>io.micrometer</groupId>
		<artifactId>micrometer-tracing-bridge-otel</artifactId>
</dependency>
<!-- Adds an exporter to store the traces -->
<dependency>
	<groupId>io.opentelemetry</groupId>
	<artifactId>opentelemetry-exporter-otlp</artifactId>
</dependency>
```

It's not clear to my how to manually instrumenting my project in order to generate mongodb traces.

Any ideas?

**Comment by jkwatson** [comment]
I'm moving this over to the instrumentation repository, as it seems to primarily be an instrumentation question.

**Comment by laurit** [comment]
@jeusdi you should ask this question from micrometer tracing community.

**Comment by trask** [comment]
another option is to use the OpenTelemetry Spring Boot starter which includes [automatic mongodb tracing](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11211) in the latest 2.4.0 release
